@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1441,22 +1441,27 @@ export class TextElementsRenderer {
     }
 
     private addPointLabel(
-        pointLabel: TextElement,
         labelState: TextElementState,
         groupState: TextElementGroupState,
         position: THREE.Vector3,
         screenPosition: THREE.Vector2,
         poiRenderer: PoiRenderer,
         textCanvas: TextCanvas,
         renderParams: RenderParams,
-        temp: TempParams
+        temp: TempParams,
+        iconIndex?: number
     ): boolean {
-        const textRenderState: RenderState = labelState.textRenderState!;
-        const iconRenderState: RenderState = labelState.iconRenderState!;
+        const pointLabel: TextElement = labelState.element;
+        const textRenderState: RenderState | undefined = labelState.textRenderState;
+        const iconRenderState: RenderState =
+            iconIndex !== undefined
+                ? labelState.iconRenderStates![iconIndex]
+                : labelState.iconRenderState!;
```

Comment:
should you check if iconRenderState is undefined?

Reply:
I'll add the check as an assertion, a point label state has always either a iconRenderStates or iconRenderState defined.